### ISSUE TITLE
react component static property has not be handled.

### ISSUE THREAD

**Issue by huyansheng3** [vicwang163/react-to-vue#4]
The  project is interesting, so i try convert [react-load-script](https://github.com/blueberryapps/react-load-script) ,but it cannot run in vue .
 
 ``` jsx
import React from 'react';
import { PropTypes as RPT } from 'prop-types';

export default class Script extends React.Component {

  static propTypes = {
    attributes: RPT.object, // eslint-disable-line react/forbid-prop-types
    onCreate: RPT.func,
    onError: RPT.func.isRequired,
    onLoad: RPT.func.isRequired,
    url: RPT.string.isRequired,
  };

  static defaultProps = {
    attributes: {},
    onCreate: () => {},
    onError: () => {},
    onLoad: () => {},
  }

  // A dictionary mapping script URLs to a dictionary mapping
  // component key to component for all components that are waiting
  // for the script to load.
  static scriptObservers = {};

  // A dictionary mapping script URL to a boolean value indicating if the script
  // has already been loaded.
  static loadedScripts = {};

  // A dictionary mapping script URL to a boolean value indicating if the script
  // has failed to load.
  static erroredScripts = {};

  // A counter used to generate a unique id for each component that uses
  // ScriptLoaderMixin.
  static idCount = 0;

  constructor(props) {
    super(props);
    this.scriptLoaderId = `id${this.constructor.idCount++}`; // eslint-disable-line space-unary-ops, no-plusplus
  }

  componentDidMount() {
    const { onError, onLoad, url } = this.props;

    if (this.constructor.loadedScripts[url]) {
      onLoad();
      return;
    }

    if (this.constructor.erroredScripts[url]) {
      onError();
      return;
    }

    // If the script is loading, add the component to the script's observers
    // and return. Otherwise, initialize the script's observers with the component
    // and start loading the script.
    if (this.constructor.scriptObservers[url]) {
      this.constructor.scriptObservers[url][this.scriptLoaderId] = this.props;
      return;
    }

    this.constructor.scriptObservers[url] = { [this.scriptLoaderId]: this.props };

    this.createScript();
  }

  componentWillUnmount() {
    const { url } = this.props;
    const observers = this.constructor.scriptObservers[url];

    // If the component is waiting for the script to load, remove the
    // component from the script's observers before unmounting the component.
    if (observers) {
      delete observers[this.scriptLoaderId];
    }
  }

  createScript() {
    const { onCreate, url, attributes } = this.props;
    const script = document.createElement('script');

    onCreate();

    // add 'data-' or non standard attributes to the script tag
    if (attributes) {
      Object.keys(attributes).forEach(prop => script.setAttribute(prop, attributes[prop]));
    }

    script.src = url;

    // default async to true if not set with custom attributes
    if (!script.hasAttribute('async')) {
      script.async = 1;
    }

    const callObserverFuncAndRemoveObserver = (shouldRemoveObserver) => {
      const observers = this.constructor.scriptObservers[url];
      Object.keys(observers).forEach((key) => {
        if (shouldRemoveObserver(observers[key])) {
          delete this.constructor.scriptObservers[url][this.scriptLoaderId];
        }
      });
    };
    script.onload = () => {
      this.constructor.loadedScripts[url] = true;
      callObserverFuncAndRemoveObserver((observer) => {
        observer.onLoad();
        return true;
      });
    };

    script.onerror = () => {
      this.constructor.erroredScripts[url] = true;
      callObserverFuncAndRemoveObserver((observer) => {
        observer.onError();
        return true;
      });
    };

    document.body.appendChild(script);
  }

  render() {
    return null;
  }
}
 
 ```

``` vue
<script>
export default {
  name: 'script',
  props: {
    attributes: {
      type: Object,
      default: {},
    },
    onCreate: {
      type: Function,
      default: () => {},
    },
    onError: {
      type: Function,
      required: true,
      default: () => {},
    },
    onLoad: {
      type: Function,
      required: true,
      default: () => {},
    },
    url: {
      type: String,
      required: true,
    },
  },
  methods: {
    createScript() {
      debugger
      const { onCreate, url, attributes } = this
      const script = document.createElement('script')
      onCreate()
      // add 'data-' or non standard attributes to the script tag
      if (attributes) {
        Object.keys(attributes).forEach(prop =>
          script.setAttribute(prop, attributes[prop])
        )
      }
      script.src = url
      // default async to true if not set with custom attributes
      if (!script.hasAttribute('async')) {
        script.async = 1
      }
      const callObserverFuncAndRemoveObserver = shouldRemoveObserver => {
        const observers = this.constructor.scriptObservers[url]
        Object.keys(observers).forEach(key => {
          if (shouldRemoveObserver(observers[key])) {
            delete this.constructor.scriptObservers[url][this.scriptLoaderId]
          }
        })
      }
      script.onload = () => {
        this.constructor.loadedScripts[url] = true
        callObserverFuncAndRemoveObserver(observer => {
          observer.onLoad()
          return true
        })
      }
      script.onerror = () => {
        this.constructor.erroredScripts[url] = true
        callObserverFuncAndRemoveObserver(observer => {
          observer.onError()
          return true
        })
      }
      document.body.appendChild(script)
    },
  },
  created() {
    this.scriptLoaderId = `id${this.constructor.idCount++}` // eslint-disable-line space-unary-ops, no-plusplus
  },
  mounted() {
    debugger
    const { onError, onLoad, url } = this
    if (this.constructor.loadedScripts[url]) {
      onLoad()
      return
    }
    if (this.constructor.erroredScripts[url]) {
      onError()
      return
    }
    // If the script is loading, add the component to the script's observers
    // and return. Otherwise, initialize the script's observers with the component
    // and start loading the script.
    if (this.constructor.scriptObservers[url]) {
      this.constructor.scriptObservers[url][this.scriptLoaderId] = this
      return
    }
    this.constructor.scriptObservers[url] = {
      [this.scriptLoaderId]: this,
    }
    this.createScript()
  },
  destroyed() {
    const { url } = this
    const observers = this.constructor.scriptObservers[url]
    // If the component is waiting for the script to load, remove the
    // component from the script's observers before unmounting the component.
    if (observers) {
      delete observers[this.scriptLoaderId]
    }
  },
  render() {
    return null
  },
}
</script>
```

in vue, this.constructor.scriptObservers is undefined.

**Comment by vicwang163** [comment]
@huyansheng3 yes, react-to-vue does not handle static variables and methods except for propTypes and defaultProps for now, and moving the static variables and methods out of react class body will be my fix, if I do this, I still need to think about the cases that how people use the static variables, e.g. 'this.contructor.xxx', 'script.xxx'. Do you have suggestion?

**Comment by huyansheng3** [comment]
@vicwang163  i am not very familiar with Vue, but in react, static properties are usually use as module export or shared (global) variables.

and es6 extends, when child extends parent, the parent static propeties will be extended by child. so maybe u should handle like es6.

add static property to VueComponent, but be sure not override the original property.

```
function _inherits(subClass, superClass) {
  if (typeof superClass !== 'function' && superClass !== null) {
    throw new TypeError(
      'Super expression must either be null or a function, not ' +
        typeof superClass
    )
  }
  subClass.prototype = Object.create(superClass && superClass.prototype, {
    constructor: {
      value: subClass,
      enumerable: false,
      writable: true,
      configurable: true,
    },
  })
  if (superClass)
    Object.setPrototypeOf
      ? Object.setPrototypeOf(subClass, superClass)
      : (subClass.__proto__ = superClass)
}
```

**Comment by vicwang163** [comment]
fix at #7
